vm/package/deploy.rs

```diff
@@ -14,31 +14,34 @@
 // You should have received a copy of the GNU General Public License
 // along with the snarkVM library. If not, see <https://www.gnu.org/licenses/>.
 
-use snarkvm_compiler::Deployment;
-use snarkvm_console::types::Address;
+use snarkvm_compiler::Transaction;
+use snarkvm_console::{
+    program::{Plaintext, Record},
+    types::Address,
+};
 
 use super::*;
 
 pub struct DeployRequest<N: Network> {
-    deployment: Deployment<N>,
+    transaction: Transaction<N>,
     address: Address<N>,
     program_id: ProgramID<N>,
 }
 
 impl<N: Network> DeployRequest<N> {
     /// Sends the request to the given endpoint.
-    pub fn new(deployment: Deployment<N>, address: Address<N>, program_id: ProgramID<N>) -> Self {
-        Self { deployment, address, program_id }
+    pub fn new(transaction: Transaction<N>, address: Address<N>, program_id: ProgramID<N>) -> Self {
+        Self { transaction, address, program_id }
```

Comment:
Add a check here that the Transaction is a deployment transaction and not an execution.

Reply:
Done! I'll do the same for #1140.